GetStream/stream-chat-android
# [CAS-1848] Fix populating mentions when editing message

https://stream-io.atlassian.net/browse/CAS-1484

### 🎯 Goal
Fix populating mentions when editing a message

### 🛠 Implementation details

1. Extracted `populateMentions` method from `SendMessage` use case
2. Used the same method for populating mentions when editing the message

### 🧪 Testing

1. Send a new message with a mention
2. Check that `Message::mentionedUsers` is not empty
3. Edit the message - either change the text or add another mention
4. Observe that `Message::mentionedUsers` contains all mentioned users

### ☑️Contributor Checklist

#### General

#### Code & documentation

### ☑️Reviewer Checklist

### 🎉 GIF

_Please provide a suitable gif that describes your work on this pull request_
